### PR TITLE
Refactors decals to an element

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

yes i tested it

## Why It's Good For The Game

hm memory hog bad yes
progress on https://github.com/tgstation/tgstation/issues/53798
## Changelog
:cl:
code: Makes decals an element
/:cl:

### CODE REVIEW

**Comment by Couls** [comment]
did you test this?

**Comment by Rohesie** [comment]
did you test this?

File: code/__DEFINES/dcs/signals.dm [hunk 1]
```diff
@@ -672,6 +672,9 @@
 	///Returned by cleanable components when they are cleaned.
 	#define COMPONENT_CLEANED	(1<<0)
 
+///from /datum/component/forensics/wipe_blood_dna()
```
Comment (Rohesie) [inline code/__DEFINES/dcs/signals.dm:675]:
```suggestion
///from base of datum/component/forensics/wipe_blood_dna() ()
```

File: code/datums/components/forensics.dm [hunk 2]
```diff
@@ -44,7 +44,7 @@
 /datum/component/forensics/proc/wipe_blood_DNA()
 	blood_DNA = null
 	if(isitem(parent))
-		qdel(parent.GetComponent(/datum/component/decal/blood))
+		SEND_SIGNAL(src, COMSIG_WIPE_BLOOD_DNA)
```
Comment (Rohesie) [inline code/datums/components/forensics.dm:47]: The signal should be generic, not dependent on whether the parent is an item or not.
Can you add the if check on whether to register to the signal in the first place instead?

**Comment by Rohesie** [comment]
Is the behavior unchanged without the item-parent check?

**Comment by TiviPlus** [comment]
yes, the registersignal checks for isitem already

**Comment by Rohesie** [comment]
Very nice, great job!

File: code/datums/components/decal.dm [hunk 3]
```diff
@@ -1,100 +1,78 @@
...
-
-/datum/component/decal/PreTransfer()
-	remove()
-
-/datum/component/decal/PostTransfer()
-	remove()
-	apply()
-
-/datum/component/decal/proc/generate_appearance(_icon, _icon_state, _dir, _layer, _color, _alpha)
+	target.add_overlay(pic)
```
Comment (AnturK) [inline code/datums/components/decal.dm:15]: Shouldn't be here.

File: code/datums/components/decal.dm [hunk 4]
```diff
@@ -1,100 +1,78 @@
...
 
-	overlay_list += pic
-
-/datum/component/decal/proc/remove(atom/thing)
-	var/atom/master = thing || parent
-	UnregisterSignal(master,COMSIG_ATOM_UPDATE_OVERLAYS)
-	master.update_icon()
-	if(isitem(master))
-		addtimer(CALLBACK(master, /obj/item/.proc/update_slot_icon), 0, TIMER_UNIQUE)
+	source.add_overlay(pic)
```
Comment (AnturK) [inline code/datums/components/decal.dm:57]: Should be `overlay_list += pic`
Overlay signal should not be adding the overlays itself, instead it should return them in overlay list.

**Review by AnturK — COMMENTED** [review]
Pretty broken

**Pull request merged**